pkg/server/definitionHandler.go

```diff
@@ -0,0 +1,244 @@
+package server
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+
+	mycue "github.com/oam-dev/kubevela/pkg/cue"
+	"github.com/oam-dev/kubevela/pkg/server/util"
+	"github.com/oam-dev/kubevela/pkg/utils/common"
+	"github.com/oam-dev/kubevela/pkg/utils/system"
+)
+
+const (
+	// OpenAPISchemaDir is the folder name under ~/.vela/capabilities
+	OpenAPISchemaDir = "openapi"
+	// UsageTag is usage comment annotation
+	UsageTag = "+usage="
+	// ShortTag is the short alias annotation
+	ShortTag = "\\n+short"
+)
+
+// OpenAPISchema is the struct for OpenAPI Schema generated by Cue OpenAPI
+type OpenAPISchema struct {
+	OpenAPI    string     `json:"openapi"`
+	Components Components `json:"components"`
+}
+
+// Components is the struct filed of OpenAPISchema
+type Components struct {
+	Schemas Schemas `json:"schemas"`
+}
+
+// Schemas is the struct filed of Components
+type Schemas struct {
+	Parameter map[string]interface{} `json:"parameter"`
+}
+
+// GetDefinition gets OpenAPI schema from Cue section of a WorkloadDefinition/TraitDefinition
+func (s *APIServer) GetDefinition(c *gin.Context) {
+	definitionName := c.Param("name")
+	parameter, err := getDefinition(definitionName)
+	if err != nil {
+		util.HandleError(c, util.StatusInternalServerError, err)
+		return
+	}
+	util.AssembleResponse(c, string(parameter), nil)
+}
+
+// getDefinition is the main function for GetDefinition API
+func getDefinition(name string) ([]byte, error) {
```

Comment:
I would point out two problems here:

1. getDefinition() directly depends on the `$HOME/.vela/caps/*.cue` files.
  This makes the server more coupled with the local Vela CLI. As moving towards a decoupled picture, the server should provide an interface to get the CUE templates instead of relying on local storage assumptions.

2. generate() copying lines starting with `parameter: {` prefix until it hits `}` end symbol.
  This is not a safe and systematic way to grab the parameter definitions. What I said in offline discussion is to only change `parameters` to cue definition since it is cue value here. Then parse it through CUE which is safe and correct.
  Let's discuss in persons if you need more clarification.

Reply:
I will hold the first suggestion until PR #776 is merged. Done for the second one. Thanks.